src/data-source/DataSource.ts

```diff
@@ -750,7 +750,10 @@ export class DataSource {
      * Get the replication mode SELECT queries should use for this datasource by default
      */
     defaultReplicationModeForReads(): ReplicationMode {
-        if ("replication" in this.driver.options) {
+        if (
+            "replication" in this.driver.options &&
```

Comment:
`this.driver.options.replication` is enough, the additional condition is unnecessary and can be removed.